lib/cc/engine/analyzers/python/node.rb

```diff
@@ -1,69 +1,28 @@
+require 'cc/engine/analyzers/node'
+
 module CC
   module Engine
     module Analyzers
       module Python
-        class Node
+        class Node < CC::Engine::Analyzers::Node
           SCRUB_PROPERTIES = ["_type", "attributes"].freeze
 
-          def initialize(node, file, default_line = 0)
-            @node = node
-            @file = file
-
-            set_default_line(default_line)
-          end
-
-          def format
-            if @node.is_a?(Hash)
-              type = @node["_type"].to_sym
-
-              if valid_properties
-                create_sexp(type, *properties_to_sexps)
-              else
-                type
-              end
-            elsif @node.is_a?(Array)
-              @node.map do |n|
-                Node.new(n, @file, @line).format
-              end
-            end
-          end
-
           private
 
-          def properties_to_sexps
-            valid_properties.map do |key, value|
-              if value.is_a?(Array) || value.is_a?(Hash)
-                create_sexp(key.to_sym, *Node.new(value, @file, @line).format)
-              else
-                create_sexp(key.to_sym, value)
-              end
-            end
-          end
+          def format_hash
+            type = @node["_type"].to_sym
 
-          def create_sexp(*args)
-            Sexp.new(*args).tap do |sexp|
-              sexp.file = @file
-              sexp.line = @line
-            end
-          end
-
-          def valid_properties
-            @node.reject do |key, value|
-              value_empty = [nil, {}, []].include?(value)
-              SCRUB_PROPERTIES.include?(key) || value_empty
-            end
-          end
-
-          def set_default_line(default)
-            if has_line_number?
-              @line = @node["attributes"]["lineno"]
+            if valid_properties
+              create_sexp(type, *properties_to_sexps)
             else
-              @line = default
+              type
             end
           end
 
-          def has_line_number?
-            @node.is_a?(Hash) && @node["attributes"] && @node["attributes"]["lineno"]
+          def line_number
+            if @node.is_a?(Hash)
+              @node.fetch("attributes", {}).fetch("lineno", nil)
```

Comment:
Turns out it raises sine `lineno` doesn't exist in `{}`. I think I'll stick to what's currently there.